index.js

```diff
@@ -1,20 +1,606 @@
+  false
+);
+
+let cup = new checkInventory(
+  "House Cup",
+  "Lucky for you the Basilisk has been dead for years - the glimmer appears to be a brilliant cup.  It appears to be the infamous House Cup! Maybe your house will win this year...",
+  "",
+  "The House Cup does not have anything to say.",
+  true
+);
+
+let mirror = new checkInventory(
+  "mirror",
+  "There is a message written in blood on the mirror.",
+  "The mirror says “The Half-Blood Prince’s Advanced Potions book belongs in the room of requirement.”",
+  "The mirror does not respond. It probably only speaks Parseltonge.",
+  false
+);
+
+let cabinet = new checkInventory(
+  "Vanishing Cabinet",
+  "You are transporting to Borgin and Burkes, where many Death Eaters await you. Game over!",
+  "The cabinet is void of words.",
+  "I would not do that if I were you.",
+  false
+);
+
+let basiliskFang = new checkInventory(
+  "Basilisk Fang",
+  "The Basilisk, also known as the King of Serpents, is a deadly beast who can kill someone with a single look directly into its piercing eyes. Its venom has only one known antidote: the tears of a phoenix.",
+  "There is nothing to read on the Basilisk Fang.",
+  "Better you than me - I would have no interest in greeting a Basilisk Fang!",
+  true
+);
+
+let sword = new checkInventory(
+  "Sword of Gryffindor",
+  "This is the sword of Godrick Gryffindor! In Harry Potter’s first year at Hogwarts, he used it to kill the basilisk snake in the Chamber of Secrets!",
+  "Just beneath the hilt of the sword, Godrick Gryffindor's name is engraved.",
+  "You cannot greet the sword.",
+  true
+);
+
+//item key - given a string input maps to the corresponding item object
+let itemKey = {
+  treacletart: treacleTart,
+  "treacle tart": treacleTart,
+  tart: treacleTart,
+  sortinghat: sortingHat,
+  "sorting hat": sortingHat,
+  hat: sortingHat,
+  fabric: cloak,
+  "invisibility cloak": cloak,
+  cloak: cloak,
+  "cloak of invisibility": cloak,
+  diary: diary,
+  "tom riddle's diary": diary,
+  "riddle's diary": diary,
+  "tom's diary": diary,
+  "riddle diary": diary,
+  "tom riddle diary": diary,
+  portrait: portrait,
+  potionbook: potionBook,
+  book: potionBook,
+  textbook: potionBook,
+  "potion book": potionBook,
+  "text book": potionBook,
+  "half-blood prince book": potionBook,
+  "halfblood prince book": potionBook,
+  "half blood prince book": potionBook,
+  scroll: scroll,
+  "chalk board": chalkBoard,
+  chalkboard: chalkBoard,
+  cup: cup,
+  "house cup": cup,
+  housecup: cup,
+  mirror: mirror,
+  cabinet: cabinet,
+  "vanishing cabinet": cabinet,
+  vanishingcabinet: cabinet,
+  sword: sword,
+  "gryffindor sword": sword,
+  "sword of gryffindor": sword,
+  basiliskfang: basiliskFang,
+  "basilisk fang": basiliskFang,
+  fang: basiliskFang,
+  inventory: player.inventory,
+  north: "north",
+  south: "south",
+  east: "east",
+  west: "west",
+  around: "around",
+  room: "roomOfRequirement",
+};
+
```

Comment:
This is a great lookup table! It looks as if you covered the majority of possible user inputs